### ISSUE TITLE
Plugin build facing issue of "plugin was built with a different version of package io"

### ISSUE THREAD

**Issue by Ivelraj1997** [lf-edge/ekuiper#2159]
**Please describe your problem in detail, if necessary, you can upload the log file through the attachment**:
kuiper latest version -1.11.0-alpha.2
Go Version -1.20 

<img width="910" alt="image" src="https://github.com/lf-edge/ekuiper/assets/37973269/e00309a2-6e0f-433b-bb66-e2a836b062cd">

build go version -1.20 
machine alpine linux

**Comment by Rui-Gan** [comment]
This appears to be a plugin compilation issue. Please refer to https://ekuiper.org/docs/en/latest/extension/native/develop/plugins_tutorial.html#plugin-development for detailed information.

**Comment by Ivelraj1997** [comment]
Thank you , It is working

**Comment by Ivelraj1997** [comment]
Just I have another query , I'm using sql predefined plugin with postgres , is there any way to pass dynamic schema ? or dynamic sink configuration?

**Comment by Rui-Gan** [comment]
Can you provide a specific example to clarify what you mean by dynamic sink configuration? If you are referring to the ability to choose different tables based on the output field, I'm afraid that feature is not currently supported.

**Comment by Ivelraj1997** [comment]
yeah you are correct , It is like to dynamically change table / database schema , it would be helpful to perform more actions  dynamically.   okay thank you for your info , hope future will get these features

**Comment by Ivelraj1997** [comment]
Another doubt , So the portable plugin python , is it possible to use dataTemplate with dynamic data passing?

**Comment by Ivelraj1997** [comment]
Okay thank you

**Comment by Ivelraj1997** [comment]
I have another query , when registering the portable plugin and the plugin contain install shell script while this getting permission issue  

<img width="656" alt="image" src="https://github.com/lf-edge/ekuiper/assets/37973269/82343ca9-139e-447d-a03b-d9bc9e4d98f8">

install.sh

set -ex
apk add --update --no-cache python3 
apk add py-pip 
pip3 install ekuiper
pip3 install redis

**Comment by ngjaying** [comment]
> I have another query , when registering the portable plugin and the plugin contain install shell script while this getting permission issue
> 
> <img alt="image" width="656" src="https://user-images.githubusercontent.com/37973269/259704676-82343ca9-139e-447d-a03b-d9bc9e4d98f8.png">
> install.sh
> 
> set -ex apk add --update --no-cache python3 apk add py-pip pip3 install ekuiper pip3 install redis

I guess you are using alpine image which does not default to root user. We recommend to use slim-python image if you need python env.

**Comment by Ivelraj1997** [comment]
OKay got it . In python to print the log 

<img width="917" alt="image" src="https://github.com/lf-edge/ekuiper/assets/37973269/14ea0990-8e5d-450a-a8ac-5e5d23bfabf6">

I have set true for the configuration value debug and console 

<img width="554" alt="image" src="https://github.com/lf-edge/ekuiper/assets/37973269/fc4f4037-9e3d-4705-8196-4876c22b6c94">

this is my code , I have added python print and using context logger function  but nothing print in the logs . is anything wrongly here?

This print added in collect interface method

**Comment by Ivelraj1997** [comment]
okay thank you

**Comment by ngjaying** [comment]
> okay thank you , Why is print logs not working inside python collect interface method?
> 
> seems like collect interface method Sometime it is get called and other time not and I have tried restart the system too
> 
> If I stop query then I can see the print log
> 
> version - lfedge/ekuiper:1.11.0-alpha.2-slim-python

Do you find any python printed logs?  Please also check the metrics to see the in/out messages count of sinks

**Comment by Ivelraj1997** [comment]
I have tried other docker image like without python environment , there it is working  , I'm not sure why is not printed in slim-python .

I have another query ,  is there possible to develop lookup custom source plugin in portable plugin development?

**Comment by ngjaying** [comment]
> I have tried other docker image like without python environment , there it is working , I'm not sure why is not printed in slim-python .
> 
> I have another query , is there possible to develop lookup custom source plugin in portable plugin development?

Sounds like a bug in slim-python.

Portable plugin does not support lookup yet.

**Comment by Ivelraj1997** [comment]
Okay . is it possible to get base/root path inside python plugin context?

**Comment by ngjaying** [comment]
> Okay . is it possible to get base/root path inside python plugin context?

In the next version (already merged into master branch), the working dir will set to the path of your plugin.

**Comment by Ivelraj1997** [comment]
okay thanks ,  In rule actions dynamically push or pop like mqtt/rest/sql based on condition , is it possible?   If not then it could be helpful in future.

**Comment by ngjaying** [comment]
Do you mean dynamic target, like dynamic MQTT topic for MQTT sink?

Multiple sinks run in parallel.

**Comment by Ivelraj1997** [comment]
No, In rule have action array right, that action array would pass dynamically actions, like  based on rule result would be either  MQTT or REST action .  

If multiple sink action either parallel or serial based on configuration option , it would be helpful , like scenario If it is serial then first action may failed the second one stop

**Comment by ngjaying** [comment]
> No, In rule have action array right, that action array would pass dynamically actions, like based on rule result would be either MQTT or REST action .
> 
> If multiple sink action either parallel or serial based on configuration option , it would be helpful , like scenario If it is serial then first action may failed the second one stop
> 
> Also in file upload getting error
> 
> <img alt="image" width="637" src="https://user-images.githubusercontent.com/37973269/261520935-7a12c95f-a7a2-409a-b701-6393986bc5e3.png">

This issue #1973 is about "serial" actions. Actually, I think this is not really a sink, the first action acts like an operator  similar to a normal function like abs(a) which returns a result for the next action. External function can wrap a REST service call as a function so that REST -> MQTT serial pipeline can be achieved.

**Comment by ngjaying** [comment]
Close as inactive. Feel free to reopen if you have more problems
